app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt

```diff
@@ -0,0 +1,56 @@
+package au.com.gridstone.trainingkotlin.adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.NonNull
+import androidx.recyclerview.widget.RecyclerView
+import au.com.gridstone.trainingkotlin.BuildConfig.IMAGE_URL
+import au.com.gridstone.trainingkotlin.R
+import au.com.gridstone.trainingkotlin.data.PokemonResults
+import au.com.gridstone.trainingkotlin.data.Pokemon
+import com.bluelinelabs.conductor.Router
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.changehandler.FadeChangeHandler
+import com.squareup.picasso.Picasso
+
+internal class HomeListAdapter(private val router: Router) :
+  RecyclerView.Adapter<HomeListAdapter.MyViewHolder>() {
+  private var itemsList: List<Pokemon> = emptyList()
+
+  internal class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    val image: ImageView = view.findViewById(R.id.image_list)
+    val name: TextView = view.findViewById(R.id.name)
+  }
+
+  @NonNull
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+    val itemView: View = LayoutInflater.from(parent.context)
+      .inflate(R.layout.item_home_recyclerview, parent, false)
+    return MyViewHolder(itemView)
+  }
+
+  override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+    val item: Pokemon = itemsList[position]
+    holder.name.text = item.name
+    loadImage(holder, item)
+    holder.itemView.setOnClickListener {
+     TODO("WORKING ON THIS IN NEXT PR #4")
+    }
+  }
+
+  private fun loadImage(holder: MyViewHolder, item: Pokemon) =
+    Picasso.get()
+      .load("$IMAGE_URL/${item.id}.png")
+      .placeholder(R.drawable.ic_placeholder)
+      .into(holder.image)
+
+  override fun getItemCount(): Int = itemsList.size
+
+  fun setList(it: PokemonResults) {
```

Comment:
if the signature reads `setList` it should probably take in a list 🤔

Reply:
sure  i set the signature to setPokemonResults